go-controller/pkg/util/subnet_annotations.go

```diff
@@ -37,82 +38,110 @@ const (
 	ovnNodeSubnets = "k8s.ovn.org/node-subnets"
 )
 
-func createSubnetAnnotation(annotationName string, defaultSubnets []*net.IPNet) (map[string]interface{}, error) {
+// updateSubnetAnnotation add the hostSubnets of the given network to the input node annotations;
+// if hostSubnets is empty, it delete the existing subnet annotation for given network from the input node annotations.
```

Comment:
if hostSubnets is empty, delete the existing subnet annotation for the given network from the input node annotations.